extensions/binding/org.eclipse.smarthome.binding.wemo/src/main/java/org/eclipse/smarthome/binding/wemo/handler/WemoLightHandler.java

```diff
@@ -0,0 +1,360 @@
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.library.types.PercentType;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusInfo;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.io.transport.upnp.UpnpIOParticipant;
+import org.eclipse.smarthome.io.transport.upnp.UpnpIOService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * {@link WemoLightHandler} is the handler for a WeMo light, responsible for handling commands and state updates for the
+ * different channels of a WeMo light.
+ *
+ * @author Hans-Jörg Merk - Initial contribution
+ */
+public class WemoLightHandler extends BaseThingHandler implements UpnpIOParticipant {
+
+    private final Logger logger = LoggerFactory.getLogger(WemoLightHandler.class);
+
+    private UpnpIOService service;
+
+    private WemoBridgeHandler wemoBridgeHandler;
+
+    private String wemoLightID;
+
+    protected final static int SUBSCRIPTION_DURATION = 600;
+
+    public final static Set<ThingTypeUID> SUPPORTED_THING_TYPES = Collections.singleton(THING_TYPE_MZ100);
+
+    /**
+     * The default refresh interval in Seconds.
+     */
+    private int DEFAULT_REFRESH_INTERVAL = 60;
+
+    private ScheduledFuture<?> refreshJob;
+
+    private Runnable refreshRunnable = new Runnable() {
+
+        @Override
+        public void run() {
+            try {
+                getDeviceState();
+            } catch (Exception e) {
+                logger.debug("Exception during poll : {}", e);
+            }
+        }
+    };
+
+    public WemoLightHandler(Thing thing, UpnpIOService upnpIOService) {
+        super(thing);
+
+        if (upnpIOService != null) {
+            logger.debug("UPnPIOService '{}'", upnpIOService);
+            this.service = upnpIOService;
+        } else {
+            logger.debug("upnpIOService not set.");
+        }
+    }
+
+    @Override
+    public void initialize() {
+
+        logger.debug("Initializing WemoLightHandler");
+
+        final String configLightId = (String) getConfig().get(DEVICE_ID);
+
+        if (DEVICE_ID != null) {
+            wemoLightID = configLightId;
+            logger.debug("Initializing WemoLightHandler for LightID '{}'", wemoLightID);
+
+            if (getWemoBridgeHandler() != null) {
+                ThingStatusInfo statusInfo = getBridge().getStatusInfo();
+                updateStatus(statusInfo.getStatus(), statusInfo.getStatusDetail(), statusInfo.getDescription());
```

Comment:
This is exactly the same like in the HueLightHandler.
So why is it ok there, but not here ??